cmd/file.d_e2e_new_test.go

```diff
@@ -0,0 +1,253 @@
+)
+
+/*
+General interface for e2e tests
+AddConfigSettings - function that preparing а config for your test
+Send - function that send message in pipeline and waits for the end of processing
+Validate - function that validate result of the work
+*/
+
+type e2eTest interface {
+	AddConfigSettings(t *testing.T, conf *cfg.Config, pipeName string)
+	Send(t *testing.T)
+	Validate(t *testing.T)
+}
+
+func startForTest(t *testing.T, e e2eTest, configPath, pipeName string) *fd.FileD {
+	conf := cfg.NewConfigFromFile(configPath)
+	e.AddConfigSettings(t, conf, pipeName)
+	filed := fd.New(conf, "off")
+	filed.Start()
+	return filed
+}
+
+func TestE2EStabilityWorkCase(t *testing.T) {
+	testsList := []struct {
+		e2eTest
+		cfgPath  string
+		pipeName string
+	}{
+		{
+			e2eTest: &fileFile{
+				outputNamePattern: "/file-d.log",
+				filePattern:       "pod_ns_container-*",
+				count:             10,
+				lines:             500,
+				retTime:           "1s",
+			},
+			cfgPath:  "./../testdata/config/e2e_cfg/file_file.yaml",
+			pipeName: "test_file_file",
+		},
+		{
+			e2eTest: &httpFile{
+				outputNamePattern: "/file-d.log",
+				count:             10,
+				lines:             500,
+				retTime:           "1s",
+			},
+			cfgPath:  "./../testdata/config/e2e_cfg/http_file.yaml",
+			pipeName: "test_http_file",
+		},
+		{
+			e2eTest: &kafkaFile{
+				topic:             "quickstart5",
+				broker:            "localhost:9092",
+				outputNamePattern: "/file-d.log",
+				lines:             500,
+				retTime:           "1s",
+				partition:         4,
+			},
+			cfgPath:  "./../testdata/config/e2e_cfg/kafka_file.yaml",
+			pipeName: "test_kafka_file",
+		},
+	}
+
+	for _, test := range testsList {
+		startForTest(t, test.e2eTest, test.cfgPath, test.pipeName)
+		test.Send(t)
+		test.Validate(t)
+	}
+}
+
+// Struct for file-file plugin e2e test
+type fileFile struct {
+	filesDir          string
+	filePattern       string
+	outputNamePattern string
+	count             int
+	lines             int
+	retTime           string
+}
+
+func (f *fileFile) AddConfigSettings(t *testing.T, conf *cfg.Config, pipeName string) {
+	f.filesDir = t.TempDir()
+	offsetsDir := t.TempDir()
+	input := conf.Pipelines[pipeName].Raw.Get("input")
+	input.Set("watching_dir", f.filesDir)
+	input.Set("filename_pattern", f.filePattern)
+	input.Set("offsets_file", filepath.Join(offsetsDir, "offsets.yaml"))
+
+	output := conf.Pipelines[pipeName].Raw.Get("output")
+	output.Set("target_file", f.filesDir+f.outputNamePattern)
+	output.Set("retention_interval", f.retTime)
+}
+
+func (f *fileFile) Send(t *testing.T) {
+	wg := &sync.WaitGroup{}
+	wg.Add(f.count)
+	for i := 0; i < f.count; i++ {
+		go func() {
+			u1 := strings.ReplaceAll(uuid.NewV4().String(), "-", "")
+			u2 := strings.ReplaceAll(uuid.NewV4().String(), "-", "")
+			name := path.Join(f.filesDir, "pod_ns_container-"+u1+u2+".log")
+			file, err := os.Create(name)
+			if err != nil {
+				assert.NoError(t, err, "unexpected error")
```

Comment:
use panic here as you don't test file creation. file creation should work in any case here